languages/go/src/main/kotlin/io/vrap/codegen/languages/go/client/ClientFileProducer.kt

```diff
@@ -0,0 +1,378 @@
+            relativePath = "$basePackageName/client.go",
+            content = """|
+                |$goGeneratedComment
+                |package $basePackageName
+                |
+                |import (
+                |    "context"
+                |    "io"
+                |    "fmt"
+                |    "net/http"
+                |    "net/url"
+                |    "path"
+                |    "runtime"
+                |    "golang.org/x/oauth2"
+                |    "golang.org/x/oauth2/clientcredentials"
+                |)
+                |
+                |// Version identifies the current library version. Should match the git tag
+                |const Version = "0.1.0"
+                |
+                |type Client struct {
+                |    httpClient *http.Client
+                |    url        *url.URL
+                |    logLevel   int
+                |    userAgent  string
+                |}
+                |
+                |type ClientConfig struct {
+                |    URL         string
+                |    Credentials *clientcredentials.Config
+                |    LogLevel    int
+                |    HTTPClient  *http.Client
+                |    UserAgent   string
+                |}
+                |
+                |// NewClient creates a new client based on the provided ClientConfig
+                |func NewClient(cfg *ClientConfig) (*Client, error) {
+                |
+                |    // If a custom httpClient is passed use that
+                |    var httpClient *http.Client
+                |    if cfg.HTTPClient != nil {
+                |        if (cfg.Credentials != nil) {
+                |            httpClient = cfg.Credentials.Client(
+                |                context.WithValue(oauth2.NoContext, oauth2.HTTPClient, cfg.HTTPClient))
+                |        } else {
+                |            httpClient = cfg.HTTPClient
+                |        }
+                |    } else {
+                |        httpClient = cfg.Credentials.Client(context.TODO())
+                |    }
+                |
+                |    var userAgent = cfg.UserAgent
+                |    if (userAgent == "") {
+                |        userAgent = GetUserAgent()
+                |    }
+                |
+                |    url, err := url.Parse(cfg.URL)
+                |    if err != nil {
+                |        return nil, err
+                |    }
+                |    client := &Client{
+                |        url:        url,
+                |        logLevel:   cfg.LogLevel,
+                |        httpClient: httpClient,
+                |        userAgent:  userAgent,
+                |    }
+                |
+                |    return client, nil
+                |}
+                |
+                |func (c* Client) createEndpoint(p string) (*url.URL, error) {
+                |    url, err := url.Parse(c.url.String())
```

Comment:
as the client will not change it's URL it would make sense to change the c.url directly to the URL type to avoid parsing it on every call.